### PR TITLE
Bug 1574651 - add tc_root_url to each repository

### PR DESCRIPTION
**Pull request by camd**
I have fixed the tests and now tested this more thoroughly.  There may be one or two areas I didn't get to.  But the main flows work well.

### CODE REVIEW

**Comment by camd** [comment]
@djmitche : Here's that branch with your changes against latest master.

**Comment by camd** [comment]
@djmitche : I had to do some updates to this.  There were new unit tests added since you initially wrote this.  And retriggers were added to Perfherder Compare Chooser as well.

File: ui/perfherder/compare/CompareTable.jsx [hunk 1]
```diff
@@ -50,22 +51,24 @@ export default class CompareTable extends React.PureComponent {
...
-      results.originalRepoName,
+      RepositoryModel.getRepo(results.originalRepoName, projects),
       true,
       times,
     );
     // retrigger new revision jobs
     this.retriggerByRevision(
       results.newRetriggerableJobId,
-      results.newRepoName,
+      RepositoryModel.getRepo(results.newRepoName, projects),
```
Comment (djmitche) [inline ui/perfherder/compare/CompareTable.jsx]: I remember there was a reason that I didn't think I could call this everywhere I needed a Repository object.  I don't remember what the reason was.  Is that not an issue here?

**Comment by djmitche** [comment]
I have another patch queued up on top of this one which is, I'm sure, rapidly bitrotting.  Plz to prioritize?

File: ui/perfherder/compare/CompareTable.jsx (hunk 1 above)
Comment (camd) [reply ui/perfherder/compare/CompareTable.jsx]: Well, you have to pass in the list of ALL repositories.  So it's a bit cumbersome compared to just using the ``currentRepo``.  In Perfherder, you may want to call against different repos.  But you're basically always on the same repo in Treeherder.  So just passing ``currentRepo`` to where you need it makes sense.

**Comment by camd** [comment]
@djmitche : Yeah, I'll get this merged (hopefully) today.  I will request some help from the sheriffs in testing this on prototype.  Thanks!

**Pull request merged**